OpenRA.Mods.Common/Traits/Multipliers/UpgradableMultiplierTrait.cs

```diff
@@ -45,7 +45,7 @@ public abstract class UpgradeMultiplierTrait : IUpgradable, IDisabledTrait, ISyn
 		protected UpgradeMultiplierTrait(UpgradeMultiplierTraitInfo info, string modifierType, string actorType)
 		{
 			if (info.Modifier.Length == 0)
-				throw new Exception("No modifiers in " + modifierType + " for " + actorType);
+				throw new ArgumentException("No modifiers in " + modifierType + " for " + actorType);
```

Comment:
Any of the three arguments could need changing here if you see this exception - I usually only specify an argument if it is the only one that could affect the check.